src/test/xmeta/test_symbol_table.cpp

```diff
@@ -110,3 +110,176 @@ TEST_CASE("Enum circular dependency")
+                Int32 property_5 { set; get; }
+            }
+        }
+    )" };
+
+    std::istringstream incorrect_property_test_idl{ R"(
+        namespace N
+        {
+            interface I
+            {
+                Int32 property_1;
+                Int32 property_1 { get; }
+                Int32 property_2 { get; set; };
+                Int32 property_2 { set; };
+            }
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    REQUIRE(reader.read(property_test_idl) == 0);
+
+    auto namespaces = reader.get_namespaces();
+    auto it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto ns = it->second;
+    REQUIRE(ns->get_namespace_bodies().size() == 1);
+    auto ns_body = ns->get_namespace_bodies()[0];
+
+    auto interfaces = ns_body->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+    REQUIRE(interfaces.find("I") != interfaces.end());
+    auto iface = interfaces.at("I");
+    auto properties = iface->get_properties();
+    REQUIRE(properties.size() == 5);
+    REQUIRE(properties[0]->get_id() == "property_1");
+    REQUIRE(properties[1]->get_id() == "property_2");
+    REQUIRE(properties[2]->get_id() == "property_3");
+    REQUIRE(properties[3]->get_id() == "property_4");
+    REQUIRE(properties[4]->get_id() == "property_5");
+    REQUIRE(properties[0]->get_get_method() != nullptr);
+    REQUIRE(properties[0]->get_get_method()->get_id() == "get_property_1");
+    REQUIRE(properties[0]->get_set_method() != nullptr);
+    REQUIRE(properties[0]->get_set_method()->get_id() == "set_property_1");
+    REQUIRE(properties[1]->get_get_method() != nullptr);
+    REQUIRE(properties[1]->get_set_method() != nullptr);
+    REQUIRE(properties[1]->get_get_method()->get_id() == "get_property_2");
+    REQUIRE(properties[1]->get_set_method()->get_id() == "set_property_2");
+    REQUIRE(properties[1]->is_get_method_declared_first());
+    REQUIRE(properties[2]->get_get_method() != nullptr);
+    REQUIRE(properties[2]->get_set_method() != nullptr);
+    REQUIRE(properties[2]->get_get_method()->get_id() == "get_property_3");
+    REQUIRE(properties[2]->get_set_method()->get_id() == "set_property_3");
+    REQUIRE(!properties[2]->is_get_method_declared_first());
+    REQUIRE(properties[3]->get_get_method() != nullptr);
+    REQUIRE(properties[3]->get_set_method() != nullptr);
+    REQUIRE(properties[3]->get_get_method()->get_id() == "get_property_4");
+    REQUIRE(properties[3]->get_set_method()->get_id() == "set_property_4");
+    REQUIRE(properties[3]->is_get_method_declared_first());
+    REQUIRE(properties[4]->get_get_method() != nullptr);
+    REQUIRE(properties[4]->get_set_method() != nullptr);
+    REQUIRE(properties[4]->get_get_method()->get_id() == "get_property_5");
+    REQUIRE(properties[4]->get_set_method()->get_id() == "set_property_5");
+    REQUIRE(!properties[4]->is_get_method_declared_first());
+
+    auto classes = ns_body->get_classes();
```

Comment:
Should check that the synthesized default interface exists, and that the property accessors on the runtime class are overriding those of the interface.